src/invariant/OrderBookIsNotCrossed.cpp

```diff
@@ -0,0 +1,239 @@
+#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "invariant/OrderBookIsNotCrossed.h"
+#include "invariant/InvariantManager.h"
+#include "ledger/LedgerTxn.h"
+#include "lib/util/format.h"
+#include "main/Application.h"
+#include "xdr/Stellar-ledger-entries.h"
+
+namespace stellar
+{
+
+std::vector<std::pair<Asset, Asset>>
+extractAssetPairs(Operation const& op, LedgerTxnDelta const& ltxd)
+{
+    switch (op.body.type())
+    {
+    case MANAGE_BUY_OFFER:
+    {
+        auto const& offer = op.body.manageBuyOfferOp();
+        return {std::pair<Asset, Asset>(offer.selling, offer.buying)};
```

Comment:
std::make_pair can infer types for arguments, so until C++17 please use that

Reply:
S

Reply:
Ok will do. Ah cool I see what you mean: https://www.codingame.com/playgrounds/2205/7-features-of-c17-that-will-simplify-your-code/template-argument-deduction-for-class-templates